### ISSUE TITLE
zip per vector or zip per 128-bit lane?

### ISSUE THREAD

**Issue by cyb70289** [xtensor-stack/xsimd#825]
For vector size > 128, xsimd `zip_lo` and `zip_hi` api zips per 128-bit lane, as described in [_mm256_unpackhi_epi8](https://www.intel.com/content/www/us/en/docs/intrinsics-guide/index.html#text=_mm256_unpackhi_epi8&ig_expand=7358).
I'm wondering if this behaviour is wanted in practice. I do see real world AVX2 code has to shuffle after unpack to get a natually zipped result.
SVE suports straightforward zipping, but to match current api, it has to do some preshuffling. It looks bad as the caller may have to shuffle it back afterwards.

IMHO it would be better to change the behaviour to be `zip per vector`, not "zip per 128-bit lane". Would like to hear comments from community.

**Comment by serge-sans-paille** [comment]
I agree that zip should do just that: zipping, and not by a given lane size... or maybe we could add a compile-time parameter to the zip function (with sensible default value) to choose the zip width (?)

**Comment by amyspark** [comment]
As remarked in #840, this change broke Krita on all platforms that ship xsimd 10 onwards. This semantic change is completely inconsistent with its [API documentation](https://xsimd.readthedocs.io/en/latest/api/data_transfer.html#_CPPv4I00E6zip_hi5batchI1T1AERK5batchI1T1AERK5batchI1T1AE).

For that reason, it would be best if this change is reverted, and cross-lane zipping were implemented with separate functions to preserve compatibility.
